pkg/providers/tinkerbell/create.go

```diff
@@ -152,7 +152,7 @@ func (p *Provider) SetupAndValidateCreateCluster(ctx context.Context, clusterSpe
 			return err
 		}
 
-		managementDatacenterConfig, err := p.providerKubectlClient.GetEksaTinkerbellDatacenterConfig(ctx, managementClusterSpec.Spec.DatacenterRef.Name, clusterSpec.ManagementCluster.KubeconfigFile, clusterSpec.Cluster.Namespace)
+		managementDatacenterConfig, err := p.providerKubectlClient.GetEksaTinkerbellDatacenterConfig(ctx, managementClusterSpec.Spec.DatacenterRef.Name, clusterSpec.ManagementCluster.KubeconfigFile, managementClusterSpec.Namespace)
```

Comment:
Do we have validation that dc needs to be with the same ns as mgmt cluster? If not, maybe we should add validation there?

Reply:
Were you able to create a cluster with two different namespaces for dc and cluster? Not sure if we have this validation for other providers, unless we do. I chose not to cover that because that wasn't what we saw the error for, but if we are specifically missing for tinkerbell I can add it. If we are missing for all the providers, might be better as a follow up.

Reply:
Looks like we are missing the following validations for tinkerbell: https://github.com/aws/eks-anywhere/blob/main/pkg/cluster/vsphere.go#L53-L68

https://github.com/aws/eks-anywhere/blob/main/pkg/cluster/tinkerbell.go#L11

However, I would vote to handle this as a separate PR then as it required adding those similar validations altogether.

Reply:
Created an issue here: https://github.com/aws/eks-anywhere/issues/6452